netket/optimizer/qgt/qgt_jacobian_pytree.py

```diff
@@ -26,17 +26,25 @@
 
 from ..linear_operator import LinearOperator, Uninitialized
 
-from .treemv_logic import mat_vec
+from .qgt_jacobian_pytree_logic import mat_vec, prepare_doks
 
 
-def QGTJacobianPyTree(vstate, *, mode, rescale_shift=False) -> "QGTJacobianT":
-    O, scale = 0, 0  # compute gradient
+def QGTJacobianPyTree(vstate, *, mode, rescale_shift=True, **kwargs) -> "QGTJacobianPyTreeT":
+    O, scale = prepare_doks(vstate._apply_fun,
+        vstate.parameters,
+        vstate.samples,
+        vstate.model_state,
+        mode,
+        rescale_shift,
+    )
+
+    _, unravel = nkjax.tree_ravel(vstate.parameters)
```

Comment:
I don't like this because we can't jit over `unravel` which would mean that `__matmul__` gets jitted in every SR step for no reason (`unravel` will always do the same thing in a given run even if it's created several times).  
I can see two solutions:
* Make `params` a field of `QGTJacobianPyTreeT`. That would give us access to the shape of the RHS PyTree in a way that can be jitted over. It's ugly though because it serves no other purpose whatsoever...
* Only implement `__matmul__` for PyTrees. In practice that would be fine, but it violates the `LinearOperator` contract...

Reply:
sorry I don't understand: why can't we declare `unravel` a static argument?

Reply:
ah i see. It's only to construct the object, right?

Reply:
We can, but it is bad for performance, because `__matmul__` would be jitted anew for every QGT object, because it uses `unravel` which is a static argument

Reply:
ah yes. Then adding `parameters` to the data class seems a good idea.
cc @inailuig, who maybe has better ideas

Reply:
An alternative would be to pass a dummy `scale` even if `rescale_shift` is not used, but then it would be wise to pass `rescale_shift` so we don't waste time on multiplying vectors with 1 all the time. Ugliness, ugliness everywhere...

Reply:
let's keep things logical for the next poor guys who'll have to look at this stuff

Reply:
I take that to mean we shouldn't do magic with `scale`, which leaves passing `params` as the solution for our problems

Reply:
I mean in any case other QGT take params inside, so... who cares.